mrtarget/modules/ChEMBL.py

```diff
@@ -206,21 +210,46 @@ def get_molecules_from_evidence(self):
                     self.target2molecule[target_id]=set()
                 self.target2molecule[target_id].add(molecule_id)
 
-    def _populate_synonyms_for_molecule(self, molecule_set):
-        #if an empty set was passed by accident, skip
-        if len(molecule_set) < 1:
-            return
-
-        url = Config.CHEMBL_MOLECULE_SET.format(';'.join(molecule_set))
-        response = requests.get(url)
-        response.raise_for_status()
-        data = response.json()
-        for i in data['molecules']:
-            if 'molecule_synonyms' in i and i['molecule_synonyms']:
+    @staticmethod
+    def _populate_synonyms_for_molecule(molecule_set, molecules_syn_dict, logger):
+        def _append_to_mol2syn(m2s_dict, molecule):
+            """if molecule has synonyms create a clean entry in m2s_dict with all synms for that chembl_id.
+            Returns either None if goes ok or the molecule chembl id if something wrong"""
+            if 'molecule_synonyms' in molecule and molecule['molecule_synonyms']:
                 synonyms = []
-                for syn in i['molecule_synonyms']:
+                for syn in molecule['molecule_synonyms']:
                     synonyms.append(syn['synonyms'])
                     synonyms.append(syn['molecule_synonym'])
                 synonyms = list(set(synonyms))
-                self.molecule2synonyms[i['molecule_chembl_id']] = synonyms
+                m2s_dict[molecule['molecule_chembl_id']] = synonyms
+                return None
+            else:
+                return molecule['molecule_chembl_id']
+
+        def _get_molecules_syns_from_url(url):
+            """get molecule list from an url. returns empty {} if something wrong"""
+            r = {}
+            try:
+                response = requests.get(url)
+                response.raise_for_status()
+                r = response.json()
+            except Exception as e:
+                logger.exception('problem downloading molecule info from url %s', str(e))
+            finally:
+                return r
+
+        molecule_set_str = ';'.join(molecule_set)
```

Comment:
`molecule_set_str` is created, but only used to test if the set is empty. The code should just test the set directly instead, and this variable removed entirely.